### PR TITLE
[ML][Pipeline]Fix jobs error on pipeline component

### PR DESCRIPTION
**Pull request by brynn-code**
Side effect pr: https://github.com/Azure/azure-sdk-for-python/pull/27019/files
'jobs' should be pop out before schema load.
Fix base path.

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/component.py [hunk 1]
```diff
@@ -328,6 +328,7 @@ def _load(
         )
         if yaml_path:
             new_instance._source_path = yaml_path
+            new_instance._base_path = context[BASE_PATH_CONTEXT_KEY]
```
Comment (D-W-) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/component.py:331]: Add a comment for this? Why not continue set `_base_path` in post load?

Comment (elliotzh) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/component.py:331]: why it only applies when yaml_path is provided?

Comment (brynn-code) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/component.py:331]: uhhh, good catch, I'll move it outer

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/pipeline_component.py [hunk 2]
```diff
@@ -375,15 +375,18 @@ def _get_telemetry_values(self, *args, **kwargs):
...
             try:
-                obj.properties.component_spec["jobs"] = PipelineComponent._resolve_sub_nodes(jobs)
+                init_params_dict["jobs"] = PipelineComponent._resolve_sub_nodes(jobs)
             except Exception:  # pylint: disable=broad-except
                 # Skip parse jobs if error exists.
                 # TODO: https://msdata.visualstudio.com/Vienna/_workitems/edit/2052262
-                obj.properties.component_spec["jobs"] = None
-        return super()._from_rest_object_to_init_params(obj)
+                pass
+        init_params_dict.update(super()._from_rest_object_to_init_params(obj))
```
Comment (elliotzh) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/pipeline_component.py:388]: NIT: maybe 
```python
init_kwargs = super().xxx
try:
  init_kwargs["jobs"] = xxx
except Exception:
  ...
return init_kwargs
```

**Review by wangchao1230 — APPROVED** [review]

**Review by elliotzh — APPROVED** [review]

**Pull request merged**